### ISSUE TITLE
Change optimizer from AdamOptimizer to MomentumOptimizer

### ISSUE THREAD

**Issue by Chicoryn** [Chicoryn/dream-go#10]
As a follow-up of #5, based on some limited testing the reason we did not reach the desired accuracy with the higher regularization coefficient does not seem to be directly related to the regularization but due to the change in optimizer to Adam which was never properly investigated.

**Comment by Chicoryn** [comment]
Alternative solution? There is an implementation in PyTorch, but none is tensorflow yet but looks fairly simple to implement.

[On the convergence of Adam and Beyond](https://openreview.net/pdf?id=ryQu7f-RZ)
